saleor/graphql/shop/mutations.py

```diff
@@ -401,18 +413,32 @@ def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
             "automatically_confirm_all_new_orders",
             "automatically_fulfill_non_shippable_gift_card",
         ]
-        site = get_site_promise(info.context).get()
-        instance = site.settings
-        update_fields = []
+        update_fields = {}
         for field in FIELDS:
-            value = data["input"].get(field)
-            if value is not None:
-                setattr(instance, field, value)
-                update_fields.append(field)
+            if field in data["input"]:
+                update_fields[field] = data["input"][field]
 
         if update_fields:
-            instance.save(update_fields=update_fields)
-        return OrderSettingsUpdate(order_settings=instance)
+            channel_models.Channel.objects.update(**update_fields)
+        channel = (
+            channel_models.Channel.objects.filter(is_active=True)
+            .order_by("slug")
+            .first()
+        )
+        if channel is None:
+            raise ValidationError(
```

Comment:
Shouldn't we put this block at the start of the mutation? If there is no channel then there is no point in doing all these actions that happen before, right?